src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java

```diff
@@ -80,8 +80,25 @@
      * @param outputStream the stream to write logs to.
      * @param outputStreamOptions if {@code CLOSE} stream should be closed in auditFinished()
      * @throws IllegalArgumentException if outputStreamOptions is null.
+     * @noinspection Deprecation
+     * @noinspectionreason We are forced to keep AutomaticBean compatability
+     *     because of maven-checkstyle-plugin.
      */
-    public XMLLogger(OutputStream outputStream, OutputStreamOptions outputStreamOptions) {
+    public XMLLogger(OutputStream outputStream,
+                     AutomaticBean.OutputStreamOptions outputStreamOptions) {
+        this(outputStream, OutputStreamOptions.valueOf(outputStreamOptions.name()));
```

Comment:
@rnveach @romani I am having trouble suppressing [this error](https://circleci-tasks-prod.s3.us-east-1.amazonaws.com/storage/artifacts/2233fdd7-79cb-481c-a64c-0df373c0d898/b8d11be6-02e2-48f3-88a2-778626a28e8a/0/home/circleci/project/target/inspection-results/Deprecation.xml?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQVFQINEOCG267PDQ%2F20230226%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230226T125514Z&X-Amz-Expires=60&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEJ3%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJHMEUCIC55VEX5VAvaEsborzVIeBY6dc5I6qxgcexQA7xgCYGnAiEAvvGbCanPXXapW%2BRYQpF5DcuZnS%2F6RE9%2FRqSEjyanugkqqwIIRhADGgwwNDU0NjY4MDY1NTYiDOSiFPVkvu%2FfcdiVpiqIAs8sKHs8ri0O69OOgGiEusO8IyQfZpgV7i0eRJjZuFZbmVZaQ8fRzZcbRXB5YQP2u28TbKr%2FOranpyyqFFR6ZJf%2F7FKDXwpZZ%2BB60vm2G7puhJj2b%2BnILQkWUAvn8kRyj0u2hjM%2BdPnnJwctlQbu8ml1RlNG8S0ADN%2B90mwMZUdr0BHVBPKnulPPutNrm%2B1M%2BbHcfODbIy0krjY9GLQdQ0%2FnKiOlmPpj%2FLJXSAKLfZPG7vM%2BQmRDWrZdxbH8wloWWrX2O1K6i8av2EIzkrKFy3vEyOQeYhK57YMndA4SAFR1uPPP8aUY1Dejp%2FdykFYTXRWdcNlzXyS9SjYksuCUAIE9gel4DvirxDCMqe2fBjqdAaK1eJQU3zaVOGaoWPeHePQfFtW0ImsqiBag%2BG58dcq9%2BucCN4W%2B1xBy3P6Mrw3C%2BKJEgQ49yAjUnzdIJoh7b64PjTixmcuwagxSeG4vpWpaxk6JcqJDttlgAJYhY8TSKip4NfA0kITJTcFrS%2FI6cZIJXXm41aGHj3gtD%2F54eU8tBbHLi5oERzHaB4AolhM7lLbtLhnq6cgbb1v2SNc%3D&X-Amz-SignedHeaders=host&x-id=GetObject&X-Amz-Signature=d3b50fb70a43cca16ba5159f7bab4b2873e181d517054ffabd9ef6bbb7ec7d46). `@noinspection Deprecation` does not work.
```xml
<problem>
  <file>
  file://$PROJECT_DIR$/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java
  </file>
  <line>88</line>
  <module>project</module>
  <package>com.puppycrawl.tools.checkstyle</package>
  <entry_point TYPE="method" FQNAME="com.puppycrawl.tools.checkstyle.XMLLoggerXMLLogger(java.io.OutputStream outputStream, com.puppycrawl.tools.checkstyle.api.AutomaticBean.OutputStreamOptions outputStreamOptions)"/>
  <problem_class id="Deprecation" severity="ERROR" attribute_key="DEPRECATED_ATTRIBUTES">Deprecated API usage</problem_class>
  <description>
  'com.puppycrawl.tools.checkstyle.api.AutomaticBean' is deprecated
  </description>
  <highlighted_element>AutomaticBean</highlighted_element>
  <language>JAVA</language>
  <offset>21</offset>
  <length>13</length>
</problem>

```

Reply:
@stoyanK7 try to place `@noinspection DeprecatedIsStillUsed` on `AutomaticBean` itself instead, let's see if that works.

Reply:
Hm, actually I noticed something. In our config file, we have `deprecation` as an allowed suppression, but actual class name is `Deprecation`. See https://github.com/checkstyle/checkstyle/blob/58130772e808e8aa95f9423962b7750043e90ccb/config/intellij-idea-inspections.xml#L4416

@stoyanK7 try to fix capitalization in allowed suppressions, maybe that will help.

Reply:
@nrmancuso I already added `Deprecation` a bit below. https://github.com/checkstyle/checkstyle/blob/3f15fd6a91d5ffc1b3cf6ad58f0b3383a8f60ee9/config/intellij-idea-inspections.xml#L4436

Reply:
@stoyanK7 also try to suppress by `@noinspection deprecation` if you haven't already :)

Reply:
Tried them both (`deprecation` and `DeprecatedIsStillUsed`) in IntelliJ, `AutomaticBean` still has a red underline. Currently running `idea-inspections.sh` to see if it says something different.

Reply:
The visual guides (red squiggly lines) seem to hang around longer than they should; you can left click in the file itself and select Analyze -> Inspect code -> File to get up to date results quicker than the script will provide.

Reply:
In 2022.2.3, `@noinspection deprecation` on XMLLogger constructor works for me.

Reply:
Running `Build 2022.3.2  #IC-223.8617.56, built on January 26, 2023`
Image is `IDEA version: ideaIC-2022.2.2`

Reply:
`deprecation` seems to remove the red line now.

Reply:
It's working. Only sonarqube is left. Thanks!

Reply:
No problem, sonarqube will have to be suppressed in web UI by maintainer after we merge PR.